src/main/java/org/skyscreamer/jsonassert/comparator/DefaultComparator.java

```diff
@@ -53,10 +53,9 @@ public void compareValues(String prefix, Object expectedValue, Object actualValu
         if (expectedValue == actualValue) {
             return;
         }
-        if ((expectedValue == null && actualValue != null) || (expectedValue != null && actualValue == null)) {
+        if (expectedValue == null || actualValue == null) {
             result.fail(prefix, expectedValue, actualValue);
-        }
-        if (areNumbers(expectedValue, actualValue)) {
+        } else if (areNumbers(expectedValue, actualValue)) {
```

Comment:
Removed extra "always true" non-null checks and do not attempt to fail for other reasons when already failing due to one value being null.